pom.xml

```diff
@@ -538,11 +538,24 @@
             <version>1.9.13</version>
         </dependency>
 
-        <!-- hadoop compilation dependency for CodecPool -->
+        <!-- hadoop compilation dependencies for CodecPool and AvroGenericRecordReader -->
         <dependency>
             <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-core</artifactId>
-            <version>0.20.2</version>
+            <artifactId>hadoop-common</artifactId>
+            <version>3.1.0</version>
+            <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
```

Comment:
I think we will also have to add the imported java files to the exclusion list of the maven shade plugin

Reply:
I do not think so as dependencies are defined as `provided`. I will double check though if those are not being shaded into uber-jar we are producing.

Reply:
Correct, `provided` dependencies should be ignored by the shade plugin.

Reply:
@rdsr is correct that we need to exclude the Avro class files in the shade configuration. The comment is not referring to these `provided` Hadoop dependencies.

Reply:
Yeah. Thanks I did not get it right.